src/Parameter.php

```diff
@@ -9,119 +9,119 @@
  * file that was distributed with this source code.
  */
 
-namespace EXSyst\Component\Swagger;
-
-use EXSyst\Component\Swagger\Parts\DescriptionPart;
-use EXSyst\Component\Swagger\Parts\ExtensionPart;
-use EXSyst\Component\Swagger\Parts\ItemsPart;
-use EXSyst\Component\Swagger\Parts\RefPart;
-use EXSyst\Component\Swagger\Parts\RequiredPart;
-use EXSyst\Component\Swagger\Parts\SchemaPart;
-use EXSyst\Component\Swagger\Parts\TypePart;
-use EXSyst\Component\Swagger\Util\MergeHelper;
-
-final class Parameter extends AbstractModel
+namespace EXSyst\OAS;
+
+final class Parameter extends AbstractObject
 {
-    use RefPart;
-    use DescriptionPart;
-    use SchemaPart;
-    use TypePart;
-    use ItemsPart;
-    use RequiredPart;
     use ExtensionPart;
 
+    const
+        IN_QUERY = 'query',
+        IN_HEADER = 'header',
+        IN_PATH = 'path',
+        IN_COOKIE = 'cookie';
+
     /** @var string */
     private $name;
 
     /** @var string */
     private $in;
 
+    /** @var string */
+    private $description;
+
+    /** @var bool */
+    private $required;
+
+    /** @var bool|null */
+    private $deprecated;
+
     /** @var bool|null */
     private $allowEmptyValue;
 
-    public function __construct($data = [])
-    {
-        $data = $this->normalize($data);
-        $this->merge($data);
+    /** @var Schema|Reference */
+    private $schema;
+
+    /** @var string */
+    private $style;
 
-        if (!$this->hasRef()) {
-            if (!isset($data['name']) || !isset($data['in'])) {
-                throw new \InvalidArgumentException('"in" and "name" are required for parameters');
-            }
+    /** @var bool */
+    private $explode;
 
-            $this->name = $data['name'];
-            $this->in = $data['in'];
-        }
-    }
+    /** @var bool */
+    private $allowReserved;
+
+    private $content; // TODO:
+
+    private $example; // TODO:
 
-    protected function doMerge($data, $overwrite = false)
+    private $examples; // TODO:
+
+    public function __construct(array $data)
     {
-        MergeHelper::mergeFields($this->allowEmptyValue, $data['allowEmptyValue'] ?? null, $overwrite);
-
-        $this->mergeDescription($data, $overwrite);
-        $this->mergeExtensions($data, $overwrite);
-        $this->mergeItems($data, $overwrite);
-        $this->mergeRef($data, $overwrite);
-        $this->mergeRequired($data, $overwrite);
-        $this->mergeSchema($data, $overwrite);
-        $this->mergeType($data, $overwrite);
+        $this->name = $data['name'];
+        $this->in = $data['in'];
+        $this->description = $data['description'] ?? null;
+        $this->required = $this->in === self::IN_PATH ? true : ($data['required'] ?? null);
```

Comment:
Can be simplified `$this->required = $this->in === self::IN_PATH ?: ($data['required'] ?? null);`

Reply:
> Determines whether this parameter is mandatory. If the parameter location is "path", this property is REQUIRED and its value MUST be true. Otherwise, the property MAY be included and its default value is false.

Reply:
I know, the code I provided behaves the same way, it just is shorter.